### DISCUSSION TITLE
Introduction of basic macros (copy of issue #79)

### DISCUSSION THREAD

**Discussion by maiixu** [SebastianMC/obsidian-custom-sort#83] [Q&A] [answered]
I have a folder note for each of my projects under each project directory. It would be nice if, instead of specifying target folder for each project, there's an option for pinning folder note, or even better, a variable such as {%foldername%} for sorting spec.

For example,
```
---
target-folder: /*
... GTD
{%foldername%}
---
```
would put foldernotes after all notes that ends with GTD.

**Comment by SebastianMC** [comment, marked as answer]
Hi @maizehsu,

Implementation completed and available in recent release ([1.8.2](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/1.8.2)) of the plugin

The final syntax is `{:%parent-folder-name%:}`, it fits seamlessly with already supported grammar and cannot be confused with a name of a real note (the `:` character is forbidden in folder and file names)

The example sorting spec reflecting your idea looks like:
```
---
sorting-spec: |
  target-folder: /*
  ... GTD
  {:%parent-folder-name%:}
---
```

**Comment by SebastianMC** [comment]
This thread is a copy of #79

**Comment by SebastianMC** [comment]
The question was [posted in Obsidian forum](https://forum.obsidian.md/t/new-plugin-obsidian-custom-sort/42706/11?u=sebastianmc) by [@RevenantsEmbrace](RevenantsEmbrace)

> Hi [@SebastianMC](https://forum.obsidian.md/u/sebastianmc) apologies for replying to this thread but I’m struggling to figure out if what I want to do is possible. the > is without the quotes as it was trying to format it.
  I have a sort spec with the below
>
>  sorting-spec: |
>  target-folder: 2. Oneshots
>  ‘>’ a-z
>  target-folder: 2. Oneshots/*
>  target-folder
>
>  i have a note in each subfolder of 2. Oneshots that is called the exact same as the folder. Is it possible to get this file first or do i have to write out each folder individually?
>
>  Any help is greatly appreciated.

The solution is to use the macro `{:%parent-folder-name%:}` as below:
```yaml
sorting-spec: |
  target-folder: 2. Oneshots
  > a-z
  target-folder: 2. Oneshots/*
  {:%parent-folder-name%:}
```

The macro `{:%parent-folder-name%:}` gets expanded to the exact name of the target folder (which is logically the parent folder for the sorted items). In result, notes with the name matching target (parent) folder will be pushed to the top. The remaining items in the folder will be sorted using default `< a-z`.

If you wish, you can explicitly specify the sorting order for the remaining items in subfolders with the indented spec like:
```yaml
sorting-spec: |
  target-folder: 2. Oneshots
  > a-z
  target-folder: 2. Oneshots/*
    > a-z
  {:%parent-folder-name%:}
```
This will sort items in subfolders with the specified reverse alphabetical order

BTW one of oncoming releases of this `custom-sort` plugin will apply the Obsidian-UI selected sorting as the default, when unspecified in spec, for the remaining items in a folder
